torchtitan/parallelisms/pipelining_utils.py

```diff
@@ -41,6 +41,18 @@ def build_pipeline_schedule(job_config, stages, loss_fn):
     if n_microbatches is None:
         n_microbatches = job_config.experimental.pipeline_parallel_degree
 
+    # Validation that the stages are compatible with the schedule
+    if isinstance(schedule_class, PipelineScheduleSingle):
```

Comment:
hm, shouldn't this validation actually go inside of PipelineSchedule* `__init__` functions?

Reply:
Hm yeah for `PipelineScheduleMulti` init we do check that `len(stages) > 2`. I guess the issue I ran into was because of this line below: https://github.com/pytorch/torchtitan/blob/ac90c36e39c6274f9beaf76922627665b6553905/torchtitan/parallelisms/pipelining_utils.py#L45

I accidentally commented out the schedule override so it defaulted to a single stage schedule, but still had the stages cut such that it was a "looped schedule" which caused a hang. So actually only the first check is necessary.

Reply:
If we wanted to be more consistent then we could just support passing in a list of stages for both single and multi, then all the validation could be done in pytorch